src/main/kotlin/uk/gov/justice/digital/hmpps/personintegrationapi/corepersonrecord/resource/CorePersonRecordV1Resource.kt

```diff
@@ -0,0 +1,199 @@
+  fun patchByPrisonerNumber(
+    @RequestParam(required = true) @Valid @ValidPrisonerNumber prisonerNumber: String,
+    @RequestBody(required = true) @Valid corePersonRecordUpdateRequest: CorePersonRecordV1UpdateRequestDto,
+  ): ResponseEntity<Void> {
+    corePersonRecordService.updateCorePersonRecordField(
+      prisonerNumber,
+      corePersonRecordUpdateRequest.fieldName,
+      corePersonRecordUpdateRequest.fieldValue,
+    )
+
+    return ResponseEntity.noContent().build()
+  }
+
+  @PutMapping(
+    "/profile-image",
+    consumes = [MediaType.MULTIPART_FORM_DATA_VALUE],
+    produces = [
+      MediaType.APPLICATION_OCTET_STREAM_VALUE,
+      MediaType.IMAGE_PNG_VALUE,
+      MediaType.IMAGE_JPEG_VALUE,
+    ],
+  )
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Add or updates the profile image on the core person record by prisoner number",
+    description = "Requires role `${CorePersonRecordRoleConstants.CORE_PERSON_RECORD_READ_WRITE_ROLE}`",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "The image file has been uploaded successfully.",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint.",
+        content = [
+          Content(
+            mediaType = MediaType.APPLICATION_JSON_VALUE,
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires ${CorePersonRecordRoleConstants.CORE_PERSON_RECORD_READ_WRITE_ROLE}.",
+        content = [
+          Content(
+            mediaType = MediaType.APPLICATION_JSON_VALUE,
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Data not found.",
+        content = [
+          Content(
+            mediaType = MediaType.APPLICATION_JSON_VALUE,
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  @PreAuthorize("hasRole('${CorePersonRecordRoleConstants.CORE_PERSON_RECORD_READ_WRITE_ROLE}')")
+  fun putProfileImageByPrisonerNumber(
+    @RequestParam(required = true) @Valid @ValidPrisonerNumber prisonerNumber: String,
+    @RequestPart(name = "imageFile", required = true) profileImage: MultipartFile,
+  ): ResponseEntity<InputStreamResource> {
+    val inputStreamResource = InputStreamResource(profileImage.inputStream)
+    return ok().contentType(
+      MediaType.parseMediaType(
+        profileImage.contentType ?: MediaType.APPLICATION_OCTET_STREAM_VALUE,
+      ),
+    ).contentLength(profileImage.size).header(
+      HttpHeaders.CONTENT_DISPOSITION,
+      "attachment; filename=\"${profileImage.originalFilename}\"",
+    ).body(inputStreamResource)
```

Comment:
Do we need to return the image back in the response?

Reply:
Good point, I've switched to a 204